File: packages/connectors/connectors/contract.py
"""Connector I/O types. Used by adapters and by apps/api runner."""

from typing import Any

from pydantic import BaseModel


class Fragment(BaseModel):
    content: str
    metadata: dict[str, Any] | None = None


class SourceMetadata(BaseModel):
    source_id: str
    type: str
    title: str
    link: str | None = None


class ConnectorOutput(BaseModel):
    success: bool
    fragments: list[Fragment]
    source_metadata: SourceMetadata
    error: str | None = None


class ConnectorLimits(BaseModel):
    max_fragments_per_source: int = 20
    max_total_fragments: int = 50
    timeout_seconds: int = 30


class ConnectorInput(BaseModel):
    query_text: str
    config: dict[str, Any]
    limits: ConnectorLimits = ConnectorLimits()

File: apps/web/lib/server/connectors/contract.ts
/**
 * Connector I/O types (align with @repo/shared). Server-only.
 */

import type { Fragment, SourceMetadata } from "@repo/shared";

export type { Fragment, SourceMetadata };

export interface ConnectorOutput {
  success: boolean;
  fragments: Fragment[];
  source_metadata: SourceMetadata;
  error: string | null;
}

export interface ConnectorLimits {
  max_fragments_per_source: number;
  max_total_fragments: number;
  timeout_seconds: number;
}

export const DEFAULT_LIMITS: ConnectorLimits = {
  max_fragments_per_source: 20,
  max_total_fragments: 50,
  timeout_seconds: 30,
};

export interface ConnectorInput {
  query_text: string;
  config: Record<string, unknown>;
  limits: ConnectorLimits;
}

export interface ConnectorAdapter {
  fetch(input: ConnectorInput): Promise<ConnectorOutput>;
  testConnection(config: Record<string, unknown>): Promise<{ ok: boolean; error?: string }>;
}
